kvdb/config.go

```diff
@@ -25,7 +25,7 @@ const (
 
 // BoltConfig holds bolt configuration.
 type BoltConfig struct {
-	SyncFreelist bool `long:"nofreelistsync" description:"Whether the databases used within lnd should sync their freelist to disk. This is disabled by default resulting in improved memory performance during operation, but with an increase in startup time."`
+	NoFreelistSync bool `long:"nofreelistsync" description:"Whether the databases used within lnd should sync their freelist to disk. This is disabled by default resulting in improved memory performance during operation, but with an increase in startup time."`
```

Comment:
We don't want to break the existing config options, the prefix may be a bit weird here, but the idea is that _not_ syncing the free list is actually the default. If users want to sync the free list (which means faster start up time, but more performance degradation with larger databases), then they need to set this flag manually.

Reply:
Ah ok, I think the issue here is actually the divergence between the command line flag itself, and how the flag is to be interpreted. So arguably, we want to do the reverse operation here: change the command line flag to match the other one.

Reply:
As is, `lnd --db.bolt.nofreelistsync`, without anything explicitly set in the config file, gives us the opposite of what is expected, namely `Opening bbolt database, sync_freelist=true, auto_compact=false`. The `--db.bolt.nofreelistsync` flag does not serve any purpose if we can not set a boolean flag to false.

_Not_ syncing is set as the default state in Incfg/db.go:
```
func DefaultDB() *DB {
	return &DB{
		Backend:             BoltBackend,
		BatchCommitInterval: DefaultBatchCommitInterval,
		Bolt: &kvdb.BoltConfig{
			NoFreelistSync:    true,
			AutoCompactMinAge: kvdb.DefaultBoltAutoCompactMinAge,
			DBTimeout:         kvdb.DefaultDBTimeout,
		},
	}
}
```

Reply:
I think it's worth changing the description too while we're here, since it's a bit confusing. "This is disabled by default" meaning `NoFreelistSync=true`. I think we could rephrase to somehting like "This is set to true by default, meaning we don't sync the free-list resulting in imporved...". WDYT?

Reply:
@bhandras Good idea. Will add a short description.